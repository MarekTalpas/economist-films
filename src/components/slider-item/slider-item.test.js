// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SliderItem from './slider-item';

const dummyItem1 = {
  title: 'Episode 6 Title',
  type: 'Episode',
  id: 6,
  subtitle: 'subtitle',
  thumbnail: {
    thumb: {
      url: 'url',
    },
  },
};

const dummyItem2 = {
  title: 'Series 1 Title',
  type: 'Series',
  id: 9,
  episode_count: 10,
  thumbnail: {
    thumb: {
      url: 'url',
    },
  },
};

const dummyItem3 = {
  title: 'Episode 19 Title',
  type: 'Episode',
  id: 19,
  subtitle: 'subtitle',
};

test('SliderItem renders correctly', () => {
  let tree : string = renderer.create(<SliderItem item={dummyItem1} />).toJSON();
  expect(tree).toMatchSnapshot();
  tree = renderer.create(<SliderItem item={dummyItem2} />).toJSON();
  expect(tree).toMatchSnapshot();
  tree = renderer.create(<SliderItem item={dummyItem3} />).toJSON();
  expect(tree).toMatchSnapshot();
});
