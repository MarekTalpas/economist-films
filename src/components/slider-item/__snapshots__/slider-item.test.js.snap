// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItem renders correctly 1`] = `
<div>
  <a
    className="slider-item"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="Episode 6 Title"
      className="slider-item-image"
      src="url"
    />
    <div
      className="slider-item-container"
    >
      <div
        className="slider-item-title"
      >
        Episode 6 Title
      </div>
      <div
        className="slider-item-subtitle"
      >
        subtitle
      </div>
    </div>
  </a>
</div>
`;

exports[`SliderItem renders correctly 2`] = `
<div>
  <a
    className="slider-item"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="Series 1 Title"
      className="slider-item-image"
      src="url"
    />
    <div
      className="slider-item-container"
    >
      <div
        className="slider-item-title"
      >
        Series 1 Title
      </div>
      <div
        className="slider-item-subtitle"
      >
        10 episodes
      </div>
    </div>
  </a>
</div>
`;

exports[`SliderItem renders correctly 3`] = `
<div>
  <a
    className="slider-item"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="Episode 19 Title"
      className="slider-item-image"
      src=""
    />
    <div
      className="slider-item-container"
    >
      <div
        className="slider-item-title"
      >
        Episode 19 Title
      </div>
      <div
        className="slider-item-subtitle"
      >
        subtitle
      </div>
    </div>
  </a>
</div>
`;
